src/gtest/test_sideChain.cpp

```diff
@@ -0,0 +1,1196 @@
+
+    ~SideChainTestSuite() {
+        sideChainManager.reset();
+    };
+
+    void SetUp() override {
+        SelectBaseParams(CBaseChainParams::REGTEST);
+        SelectParams(CBaseChainParams::REGTEST);
+
+        sideChainManager.initialUpdateFromDb(0, true, Sidechain::ScMgr::mock);
+    };
+
+    void TearDown() override {
+        resetParams();
+        resetBaseParams();
+    };
+
+protected:
+    //Subjects under test
+    Sidechain::ScMgr&           sideChainManager;
+    Sidechain::ScCoinsViewCache coinViewCache;
+
+    //Helpers
+    CBlock              aBlock;
+    CTransaction        aTransaction;
+    int                 anHeight;
+    CValidationState    txState;
+
+    CFeeRate   aFeeRate;
+    CTxMemPool aMemPool;
+    CBlockUndo aBlockUndo;
+
+    void resetBaseParams();
+    void resetParams();
+
+    void preFillSidechainsCollection();
+
+    CTransaction createSideChainTxWith(const uint256 & newScId, const CAmount & fwdTxAmount);
+    CTransaction createFwdTransferTxWith(const uint256 & newScId, const CAmount & fwdTxAmount);
+
+    CTransaction createEmptyScTx();
+    CTransaction createSideChainTxWithNoFwdTransfer(const uint256 & newScId);
+    CTransaction createNonScTx(bool ccIsNull = true);
+    CTransaction createShieldedTx();
+    void         extendTransaction(CTransaction & tx, const uint256 & scId, const CAmount & amount);
+
+    CBlockUndo   createBlockUndoWith(const uint256 & scId, int height, CAmount amount);
+    CBlockUndo   createEmptyBlockUndo();
+};
+
+///////////////////////////////////////////////////////////////////////////////
+/////////////////////////// checkTxSemanticValidity ///////////////////////////
+///////////////////////////////////////////////////////////////////////////////
+TEST_F(SideChainTestSuite, NonSideChain_CcNull_TxsAreSemanticallyValid) {
+    aTransaction = createNonScTx();
+
+    //prerequisites
+    ASSERT_FALSE(aTransaction.IsScVersion())<<"Test requires non sidechain tx";
+    ASSERT_TRUE(aTransaction.ccIsNull())<<"Test requires null tx";
+    ASSERT_TRUE(txState.IsValid())<<"Test require transition state to be valid a-priori";
+
+    //test
+    bool res = sideChainManager.checkTxSemanticValidity(aTransaction, txState);
+
+    //checks
+    EXPECT_TRUE(res);
+    EXPECT_TRUE(txState.IsValid());
+}
+
+TEST_F(SideChainTestSuite, NonSideChain_NonCcNull_TxsAreNotSemanticallyValid) {
+    aTransaction = createNonScTx(/*ccIsNull = */false);
+
+    //prerequisites
+    ASSERT_FALSE(aTransaction.IsScVersion())<<"Test requires non sidechain tx";
+    ASSERT_FALSE(aTransaction.ccIsNull())<<"Test requires null tx";
+    ASSERT_TRUE(txState.IsValid())<<"Test require transition state to be valid a-priori";
+
+    //test
+    bool res = sideChainManager.checkTxSemanticValidity(aTransaction, txState);
+
+    //checks
+    EXPECT_FALSE(res);
+    EXPECT_FALSE(txState.IsValid());
+    EXPECT_TRUE(txState.GetRejectCode() == REJECT_INVALID)
```

Comment:
It could be useful also testing the specific error tag returned, but before doing that, code base should be improved with more specific tags.
Same comment applies for other lines below.

Reply:
Agree. As discussed off-line, I am at your convenience to extend tests once we introduce specific tags in codebase